src/robusta/utils/cluster_provider_discovery.py

```diff
@@ -0,0 +1,120 @@
+import logging
+import re
+from enum import Enum, auto
+from typing import Dict, Optional
+
+from hikaru.model import NodeList
+
+
+class ClusterProviderType(Enum):
+    GKE = "GKE"
+    AKS = "AKS"
+    EKS = "EKS"
+    Kind = "Kind"
+    Minikube = "Minikube"
+    RancherDesktop = "RancherDesktop"
+    Kapsule = "Kapsule"
+    Kops = "Kops"
+    DigitalOcean = "DigitalOcean"
+    Unknown = auto()
+
+# the value is a regex match of the hostname
+HOSTNAME_MATCH: Dict[ClusterProviderType, str] = {
+    ClusterProviderType.Kind: ".*kind.*",
+    ClusterProviderType.RancherDesktop: ".*rancher-desktop.*"
+}
+
+# the value is a node label unique to the provider
+NODE_LABELS: Dict[ClusterProviderType, str] = {
+    ClusterProviderType.Minikube: "minikube.k8s.io/name",
+    ClusterProviderType.DigitalOcean: "doks.digitalocean.com/version",
+    ClusterProviderType.Kops: "kops.k8s.io/instancegroup",
+    ClusterProviderType.Kapsule: "k8s.scaleway.com/kapsule"
+}
+
+__CLUSTER_PROVIDER__ = None
```

Comment:
Here I propose not to use a global variable, that is changed in other functions of this module, but rather to have a class, manage everything in it and then to have a global variable with an instance of this class - so the logic is encapsulated in that class

Reply:
If to call that class something like "ClusterProviderDetector" it will make sense that we have a static instance of that in module
This way we could avoid using `global` actually